services/networking/loadbalancer/taxonomy.md

```diff
@@ -0,0 +1,67 @@
+# Load Balancer Taxonomy
+
+This _service-level taxonomy_ documents the minimual set of features
+that should be present for a service to be considered portable for
+use in financial services ecosystems.
+
+## Taxonomy
+
+| Taxonomy ID | Feature                                               | Description                                                                                                                                                                                                                                                                                                                   |
+| ----------- | ----------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| CCC-050301  | Application Load Balancing                            | Provide Application Load Balancers functioning at the Application Layer (Layer 7) of the OSI model. These load balancers will intelligently distribute incoming traffic based on application-level information, such as HTTP headers, cookies, or URLs, to ensure efficient and scalable application delivery.                |
```

Comment:
Should we have this as
Application Load Balancing for HTTP
Application Load Balancing for HTTPS
Then the validations would be more meaningful

Reply:
I agree it would make the validations more meaningful.

As a side note, I can only find support for `Gateway Load Balancing` via AWS, and only they specifically call it that. I've also taken the names `Application` and `Network` from AWS. Azure calls it something else entirely (`Application Gateway, Front Door, Traffic Manager, and just 'Load Balancer`). I think I should rename all of these to `Layer 7` `Layer 4` and `Layer 3` to be more name-agnostic (I think this is a suggestion you made before?)

If we go with your approach of coupling a load balancing layer with a protocol (and the layer name change I'm proposing), we'd end up with something like this:

`Layer 7 Load Balancing - HTTP/1.1`
`Layer 7 Load Balancing - HTTPS/1.1`
`Layer 7 Load Balancing - HTTP/2 / gRPC`
`Layer 7 Load Balancing - HTTP/3`
`Layer 7 Load Balancing - TCP`
`Layer 7 Load Balancing - DNS`
`Layer 4 Load Balancing - TCP`
`Layer 4 Load Balancing - SSL`
`Layer 4 Load Balancing - UDP`
`Layer 4 Load Balancing - TLS`
`Layer 3 Load Balancing - IP`

My hesitation to couple was because we could end up with lots of variations of layers and protocols any time a provider deviates from what's already listed. But now that I've seen it written out I quite like it! 

Is this what you had in mind?

Reply:
Yes - looks great! That means you can validate each one separately, it's easy to add more items as they're independent of the rest, and you can pick & choose exactly which ones are applicable to your requirements. Good job!

Reply:
Et voilà: [17d2085](https://github.com/cfisher-scottlogic/common-cloud-controls/pull/1/commits/17d2085480b8e85b318c2f47a9dcdc15578a83bd)